tensorrt_llm/runtime/kv_cache_manager_v2/_core/_moving_average.py

```
@@ -54,3 +56,40 @@ def update(self, value: int | float) -> None:
+
+
+class TimeWeightedAverage:
+    # Time-weighted lifetime mean of a piecewise-constant value. Each call to
+    # update() closes the prior segment by adding current_value * (now - t_last)
+    # to the integral, then sets current_value to the new sample. value reports
+    # integral / (now - t_init), including the still-open segment up to now.
+    # This matches the natural "average history length across the cache's life"
+    # semantic and is immune to commit-driven bursts of update() calls — many
+    # rapid samples contribute proportionally little because little wall time
+    # elapses between them.
+    __slots__ = ("integral", "current_value", "t_init", "t_last")
```

Ruff (0.15.13)

[warning] `TimeWeightedAverage.__slots__` is not sorted

Apply a natural sort to `TimeWeightedAverage.__slots__`

(RUF023)